hibernate-core/src/main/java/org/hibernate/mapping/DenormalizedTable.java

```diff
@@ -70,10 +69,12 @@ public void createForeignKeys(MetadataBuildingContext context) {
 			if ( foreignKey.getReferencedTable() == null ) {
 				foreignKey.setReferencedTable( referencedClass.getTable() );
 			}
+
+			ForeignKey normalizedForeignKey = createNormalizedForeignKey(foreignKey);
```

Comment:
Shouldn't it be

```suggestion
			ForeignKey denormalizedForeignKey = createDenormalizedForeignKey(foreignKey);
```

since we are in fact adding foreign keys?